# Analyse exploratoire de la donnée

## Importation des packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Importation de la donnée

In [2]:
database = pd.read_csv("en.openfoodfacts.org.products.tsv", delimiter='\t', low_memory=False)

In [3]:
database.shape

(356027, 163)

Le jeu de données comporte 356 027 lignes et 163 colonnes.

In [4]:
pd.set_option('display.max_columns', None)
database.head()

code                                                url  \
0  0000000003087  http://world-en.openfoodfacts.org/product/0000...   
1  0000000004530  http://world-en.openfoodfacts.org/product/0000...   
2  0000000004559  http://world-en.openfoodfacts.org/product/0000...   
3  0000000016087  http://world-en.openfoodfacts.org/product/0000...   
4  0000000016094  http://world-en.openfoodfacts.org/product/0000...   

                      creator   created_t      created_datetime  \
0  openfoodfacts-contributors  1474103866  2016-09-17T09:17:46Z   
1             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
2             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
3             usda-ndb-import  1489055731  2017-03-09T10:35:31Z   
4             usda-ndb-import  1489055653  2017-03-09T10:34:13Z   

  last_modified_t last_modified_datetime                    product_name  \
0      1474103893   2016-09-17T09:18:13Z              Farine de blé noir   
1      1489069957   2017-03-09T14:32:37Z  Banana Chips Sweetened (Whole)   
2      1489069957   2017-03-09T14:32:37Z                         Peanuts   
3      1489055731   2017-03-09T10:35:31Z          Organic Salted Nut Mix   
4      1489055653   2017-03-09T10:34:13Z                 Organic Polenta   

  generic_name quantity packaging packaging_tags           brands  \
0          NaN      1kg       NaN            NaN  Ferme t'y R'nao   
1          NaN      NaN       NaN            NaN              NaN   
2          NaN      NaN       NaN            NaN   Torn & Glasser   
3          NaN      NaN       NaN            NaN        Grizzlies   
4          NaN      NaN       NaN            NaN   Bob's Red Mill   

       brands_tags categories categories_tags categories_en origins  \
0  ferme-t-y-r-nao        NaN             NaN           NaN     NaN   
1              NaN        NaN             NaN           NaN     NaN   
2     torn-glasser        NaN             NaN           NaN     NaN   
3        grizzlies        NaN             NaN           NaN     NaN   
4   bob-s-red-mill        NaN             NaN           NaN     NaN   

  origins_tags manufacturing_places manufacturing_places_tags labels  \
0          NaN                  NaN                       NaN    NaN   
1          NaN                  NaN                       NaN    NaN   
2          NaN                  NaN                       NaN    NaN   
3          NaN                  NaN                       NaN    NaN   
4          NaN                  NaN                       NaN    NaN   

  labels_tags labels_en emb_codes emb_codes_tags first_packaging_code_geo  \
0         NaN       NaN       NaN            NaN                      NaN   
1         NaN       NaN       NaN            NaN                      NaN   
2         NaN       NaN       NaN            NaN                      NaN   
3         NaN       NaN       NaN            NaN                      NaN   
4         NaN       NaN       NaN            NaN                      NaN   

  cities cities_tags purchase_places stores countries    countries_tags  \
0    NaN         NaN             NaN    NaN     en:FR         en:france   
1    NaN         NaN             NaN    NaN        US  en:united-states   
2    NaN         NaN             NaN    NaN        US  en:united-states   
3    NaN         NaN             NaN    NaN        US  en:united-states   
4    NaN         NaN             NaN    NaN        US  en:united-states   

    countries_en                                   ingredients_text allergens  \
0         France                                                NaN       NaN   
1  United States  Bananas, vegetable oil (coconut oil, corn oil ...       NaN   
2  United States  Peanuts, wheat flour, sugar, rice flour, tapio...       NaN   
3  United States  Organic hazelnuts, organic cashews, organic wa...       NaN   
4  United States                                    Organic polenta       NaN   

  allergens_en traces traces_tags traces_en     s

In [13]:
pd.set_option('display.max_rows', None)
database.dtypes

code                                           object
url                                            object
creator                                        object
created_t                                      object
created_datetime                               object
last_modified_t                                object
last_modified_datetime                         object
product_name                                   object
generic_name                                   object
quantity                                       object
packaging                                      object
packaging_tags                                 object
brands                                         object
brands_tags                                    object
categories                                     object
categories_tags                                object
categories_en                                  object
origins                                        object
origins_tags                

## Valeurs manquantes

In [5]:
missing_values = database.isnull().sum()
print(missing_values.sum())

44183489


In [9]:
missing_percent = missing_values/len(database)*100

In [11]:
missing_percent.sort_values(0, False)

water-hardness_100g                           100.000000
no_nutriments                                 100.000000
-lignoceric-acid_100g                         100.000000
-cerotic-acid_100g                            100.000000
-melissic-acid_100g                           100.000000
glycemic-index_100g                           100.000000
-elaidic-acid_100g                            100.000000
-mead-acid_100g                               100.000000
-erucic-acid_100g                             100.000000
-nervonic-acid_100g                           100.000000
-caproic-acid_100g                            100.000000
-butyric-acid_100g                            100.000000
nutrition_grade_uk                            100.000000
ingredients_from_palm_oil                     100.000000
ingredients_that_may_be_from_palm_oil         100.000000
chlorophyl_100g                               100.000000
-palmitic-acid_100g                            99.999719
-stearic-acid_100g             

Il y a de nombreuses colonnes qui ne sont pas remplies ou peu remplies.

## Valeurs uniques dans la colonne 'countries'

In [14]:
print(len(database['countries'].unique()))

1638
